File: src/leetcode/merge-strings-alternately/merge-strings-alternately.js
/**
 * Leetcode #1768
 * Difficulty: Easy
 * https://leetcode.com/problems/merge-strings-alternately/
 */
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
export const mergeAlternately = (word1, word2) => {
  const length = Math.max(word1.length, word2.length);
  let ans = '';

  for (let i = 0; i < length; i += 1) {
    if (word1[i]) {
      ans += word1[i];
    }

    if (word2[i]) {
      ans += word2[i];
    }
  }

  return ans;
};

File: src/leetcode/merge-strings-alternately/merge-strings-alternately.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut ans: String = String::from("");
        let mut idx = 0;
        let len1 = word1.len();
        let len2 = word2.len();

        while idx < len1 || idx < len2 {
            if idx < len1 {
                ans.push(word1.chars().nth(idx).unwrap());
            }

            if idx < len2 {
                ans.push(word2.chars().nth(idx).unwrap());
            }

            idx += 1;
        }

        ans
    }
}

// impl Solution {
//     pub fn merge_alternately(word1: String, word2: String) -> String {
//         let mut ans: String = String::from("");
//         let len = usize::max(word1.len(), word2.len());
//         let word1 = word1.chars();
//         let word2 = word2.chars();
//
//         for idx in 0..len {
//             if let Some(value1) = word1.clone().nth(idx) {
//                 ans.push_str(&*value1.to_string());
//             }
//
//             if let Some(value2) = word2.clone().nth(idx) {
//                 ans.push_str(&*value2.to_string());
//             }
//         }
//
//         ans
//     }
// }